### PR TITLE
Obstacle Distance Message 

### PR DESCRIPTION
**Pull request by mrivi**
@dogmaphobic @julianoes

### CODE REVIEW

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -4397,5 +4397,12 @@
       <field type="uint8_t" name="param_type" enum="MAV_PARAM_EXT_TYPE">Parameter type: see the MAV_PARAM_EXT_TYPE enum for supported data types.</field>
       <field type="uint8_t" name="param_result" enum="PARAM_ACK">Result code: see the PARAM_ACK enum for possible codes.</field>
     </message>
+    <message id="325" name="OBSTACLE_DISTANCE" >
+      <description>Obstacle distances in front of the camera, starting from the left in increment degrees to the right</description>
+      <field type="uint64_t" name="time_usec" units="us">Timestamp (microseconds since system boot or since UNIX epoch)</field>
+      <field type="uint8_t" name="estimator_type" enum="MAV_DISTANCE_SENSOR">Class id of the distance sensor type.</field>
+      <field type="uint8_t[12]" name="distances">Distance of obstacles in front of the camera staring on the left side. A value of 0 means no obstacle. In a array element, each unit corresponds to 1m.</field>
```
Comment (julianoes) [inline message_definitions/v1.0/common.xml]: What about also having a value for unknown such as `255` ?

Comment (mrivi) [reply message_definitions/v1.0/common.xml]: I am not sure I get what you mean. 255 to indicate no obstacle in a certain array element?

Comment (julianoes) [reply message_definitions/v1.0/common.xml]: What value would you send if you don't know if there is an obstacle, so you're uncertain.

Comment (lbegani) [inline message_definitions/v1.0/common.xml]: > staring on the left side

Typo?

Comment (lbegani) [inline message_definitions/v1.0/common.xml]: > front of the camera

Should the term be camera or sensor?

File: message_definitions/v1.0/common.xml [hunk 2]
```diff
@@ -4397,5 +4397,12 @@
       <field type="uint8_t" name="param_type" enum="MAV_PARAM_EXT_TYPE">Parameter type: see the MAV_PARAM_EXT_TYPE enum for supported data types.</field>
       <field type="uint8_t" name="param_result" enum="PARAM_ACK">Result code: see the PARAM_ACK enum for possible codes.</field>
     </message>
+    <message id="325" name="OBSTACLE_DISTANCE" >
```
Comment (lbegani) [inline message_definitions/v1.0/common.xml]: Is it possible to have multiple sensors mounted to detect obstacles? Then, how to identify the sensor that this message corresponds to?

Comment (mrivi) [reply message_definitions/v1.0/common.xml]: My idea was to have this message to show already aggregated data from multiple sensors. @julianoes what do you think?

Comment (julianoes) [reply message_definitions/v1.0/common.xml]: @lbegani several sensors could send this message. You can then identify who sent it by the mavlink component ID.

Comment (lbegani) [reply message_definitions/v1.0/common.xml]: Sounds good.

**Pull request merged**